manage.py

```diff
@@ -24,76 +26,93 @@
-    parser = ArgumentParser()
-    subparsers = parser.add_subparsers(title='system', description='edX service to run')
-
-    lms = subparsers.add_parser(
-        'lms',
-        help='Learning Management System',
-        add_help=False,
-        usage='%(prog)s [options] ...'
+def main():
+    """
+    Call the management command.
+
+    Convert from legacy style into standard idiomatic django style if necessary.
+    """
+
+    env_settings_module = os.environ.get("DJANGO_SETTINGS_MODULE")
+
+    # Final args and settings: We'll determine these based on whether we're using legacy usage or the new usage...
+    final_settings_module: str
+    final_args: list[str]
+
+    if len(sys.argv) > 1 and sys.argv[1] in ["lms", "cms"]:
+        # LEGACY USAGE:
+        # The first arg after 'manage.py' is the service_variant, either 'lms' or 'cms'.
+        # Additionally, the '--settings' flag may be provided, which points to module within {service_variant}.envs.
+        # We remove both service_variant and --settings, but use them to determine a DJANGO_SETTINGS_MODULE.
+        # For example:
+        #    ./manage.py cms migrate --foo bar --settings blah --baz quux
+        # Gets converted to:
+        #    DJANGO_SETTINGS_MODULE=cms.envs.blah ./manage.py migrate --foo bar --baz quux
+        manage_py = sys.argv[0]  # ./manage.py, manage.py, or path/to/manage.py
+        service_variant = sys.argv[1]  # lms or cms
+        settings_and_management_args = sys.argv[2:]  # everything else
+
+        # Parse --settings, if it's there.
+        parse_settings_module = ArgumentParser()
+        parse_settings_module.add_argument(
+            '--settings',
+            help=(
+                f"Which django settings module to use under {service_variant}.envs. "
+                f"If unspecified, will default to {service_variant}.envs.devstack."
+            ),
+        )
+        settings_arg, management_args = parse_settings_module.parse_known_args(settings_and_management_args)
+
+        # Compute final args and settings.
+        # Order of precedence: --settings, $DJANGO_SETTINGS_MODULE, $EDX_PLATFORM_SETTINGS, devstack
+        final_settings_module = (
+            f"{service_variant}.envs.{settings_arg.settings}"
+            if settings_arg.settings
+            else (
+                f"{service_variant}.envs.{env_settings_name}"
+                if (env_settings_name := os.environ.get("EDX_PLATFORM_SETTINGS")) and not env_settings_module
+                else (
+                    env_settings_module
+                    if env_settings_module
+                    else f"{service_variant}.envs.devstack"
+                )
+            )
+        )
+        final_args = [manage_py, *management_args]
+
+    else:
+        # STANDARD IDIOMATIC DJANGO USAGE (new):
+        # The first arg after 'manage.py' is a management command... anything other than 'lms' or 'cms'.
+        # The '--settings' flag is not accepted.
```

Comment:
This seems weird to me, because, in standard Django environments it will accept either `--settings` or the `DJANGO_SETTINGS_MODULE` override.  Perhaps if `lms/cms` is the first param, then we truncate the settings flag otherwise we leave it alone and let Django handle it like normal?